cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -97,6 +103,46 @@ public static void validateConfig(SecureStore secureStore, SourceControlConfig s
     }
   }
 
+  /**
+   * Returns the <a href="https://git-scm.com/docs/git-hash-object">Git Hash</a>
+   * of the requested file path in the current HEAD. For symlinks, it returns the hash of
+   * the target file. This ensures that the file hash of a symlink is equal to the hash of the target file. It
+   * doesn't return the hash for directories. If the directory contains symlinks, we can't depend on the
+   * directory hash to ensure the contents are unchanged.
+   *
+   * @param path The absolute path on the filesystem that should include {@link RepositoryManager#getBasePath()} as a
+   *             prefix.
+   * @return The git file hash of the requested file.
+   * @throws IOException              when file isn't found, there are circular symlinks or other file IO errors.
+   * @throws IllegalArgumentException when the provided file isn't a regular file, eg: directory.
+   * @throws NotFoundException        when the file isn't committed to Git.
+   * @throws IllegalStateException    when {@link RepositoryManager#startUp()} isn't called before this.
+   */
+  public String getFileHash(Path path) throws IOException, NotFoundException {
+    validateInitialized();
+    // Git considers symlinks as separate files. So the hash of the symlink will be different from the target file.
+    // Resolve path in case of symlinks. This ensures the hash of the target file is returned.
+    path = path.toRealPath();
```

Comment:
Which commit is working copy at this point? This resolution should be done with working point at the proper revision

Reply:
The signature of the getFileHashMethod is changed now. It accepts a commit hash that will be resolved to get the revision.